rosflight/src/mavrosflight/config_manager.cpp

```diff
@@ -0,0 +1,299 @@
+{
+  comm_->register_mavlink_listener(this);
+}
+
+mavrosflight::ConfigManager::~ConfigManager()
+{
+  for (config_promise_t *promise:promises_)
+    delete promise;
+}
+
+void mavrosflight::ConfigManager::handle_mavlink_message(const mavlink_message_t &msg)
+{
+  switch (msg.msgid)
+  {
+    case MAVLINK_MSG_ID_ROSFLIGHT_CONFIG:
+      handle_config_message(msg);
+      break;
+    case MAVLINK_MSG_ID_ROSFLIGHT_DEVICE_INFO:
+      handle_device_info_message(msg);
+      break;
+    case MAVLINK_MSG_ID_ROSFLIGHT_CONFIG_INFO:
+      handle_config_info_message(msg);
+      break;
+    case MAVLINK_MSG_ID_ROSFLIGHT_CONFIG_STATUS:
+      handle_config_response_message(msg);
+      break;
+    default:
+      break;
+  }
+}
+
+void mavrosflight::ConfigManager::handle_config_message(const mavlink_message_t &msg)
+{
+  mavlink_rosflight_config_t config_msg;
+  mavlink_msg_rosflight_config_decode(&msg, &config_msg);
+  uint8_t device = config_msg.device;
+  uint8_t config = config_msg.config;
+  for (size_t promise_index{0}; promise_index < promises_.size(); promise_index++)
+    if (promises_[promise_index]->device == device)
+    {
+      promises_[promise_index]->promise.set_value(config);
+      promises_.erase(promises_.begin() + promise_index);
+      break;
+    }
+}
+
+std::tuple<bool, uint8_t> mavrosflight::ConfigManager::get_configuration(uint8_t device)
+{
+  config_promise_t *promise = new config_promise_t;
+  promise->device = device;
+  promises_.push_back(promise);
+
+  std::future<uint8_t> future = promise->promise.get_future();
+  send_config_get_request(device);
+  std::future_status result = future.wait_for(timeout);
+  bool success;
+  uint8_t config{0};
+  if (result == std::future_status::ready)
+  {
+    config = future.get();
+    success = true;
+  } else
+  {
+    success = false;
+    for(size_t promise_index{0}; promise_index < promises_.size(); promise_index++)
+      if(promises_[promise_index]==promise)
+      {
+        promises_.erase(promises_.begin() + promise_index);
+        break;
+      }
+  }
+  delete promise;
+  return std::make_tuple(success, config);
+}
+
+void mavrosflight::ConfigManager::send_config_get_request(uint8_t device)
+{
+  mavlink_message_t config_request_message;
+  mavlink_msg_rosflight_config_request_pack(1, 0, &config_request_message, device);
+  comm_->send_message(config_request_message);
+}
+
+mavrosflight::ConfigManager::config_response_t
+mavrosflight::ConfigManager::set_configuration(uint8_t device, uint8_t config)
+{
+  std::cout << "Starting set_configuration(" << +device << ", " << +config << std::endl;
+  config_response_promise_t *promise = new config_response_promise_t;
+  promise->device = device;
+  promise->config = config;
+  std::future<config_response_t> future = promise->promise.get_future();
+  config_response_promises_.push_back(promise);
+  std::cout << "Promise made. Sending request" << std::endl;
```

Comment:
Let's either get rid of all these `cout` lines, or convert them to `ROS_INFO` etc. if they're useful